src/runtime_src/core/common/drv/kds_core.c

```diff
@@ -93,13 +93,39 @@ ssize_t show_kds_scustat_raw(struct kds_sched *kds, char *buf)
 	for (i = 0; i < scu_mgmt->num_cus; ++i) {
 		sz += scnprintf(buf+sz, PAGE_SIZE - sz, cu_fmt, i,
 				scu_mgmt->name[i], scu_mgmt->status[i],
-				scu_mgmt->usage[i]);
+				scu_mgmt->usage[i], 
+				scu_mgmt->sc_usages[i].succ_cnt,
+				scu_mgmt->sc_usages[i].err_cnt,
+				scu_mgmt->sc_usages[i].crsh_cnt);
 	}
 	mutex_unlock(&scu_mgmt->lock);
 
 	return sz;
 }
 
+ssize_t kds_sk_memstat(struct kds_sched *kds, struct sk_mem_stats *stat)
+{
+	struct kds_scu_mgmt *scu_mgmt = &kds->scu_mgmt;
+	struct sk_mem_stats *mem_stat;
+
+	mutex_lock(&scu_mgmt->lock);
+
+	mem_stat = &scu_mgmt->sk_mem_stats;
+	
+	/* Check if this is present. Non Scu case fail here */
+	if (!scu_mgmt && !mem_stat) 
```

Comment:
why is this check here? we are anyway taking address of a member variable. ?

Reply:
Removed this check.